modules/core/src/main/java/org/apache/fluo/core/impl/TransactionImpl.java

```diff
@@ -596,9 +597,14 @@ public CommitData createCommitData() {
 
   @Override
   public synchronized void commit() throws CommitException {
-    SyncCommitObserver sco = new SyncCommitObserver();
-    commitAsync(sco);
-    sco.waitForCommit();
+    SyncCommitObserver sco = null;
+    try {
+      sco = new SyncCommitObserver();
+      commitAsync(sco);
+      sco.waitForCommit();
+    } finally {
+      sco = null;
```

Comment:
I wasn't sure if this was necessary, so looked into it and found this [SOQ](http://stackoverflow.com/questions/473685/does-it-help-gc-to-null-local-variables-in-java).

For this issue I think nulling `updates`, `weakNotifications`, `observedColumns`, and  `columnsRead` would be good. These can be found starting at [TransactionImpl line 109](https://github.com/apache/incubator-fluo/blob/a1c869ca508a020d028f8942225100523d5379a2/modules/core/src/main/java/org/apache/fluo/core/impl/TransactionImpl.java#L109).